src/managers/builtin/pipUtils.ts

```diff
@@ -13,21 +13,46 @@ import { Installable } from '../common/types';
 import { mergePackages } from '../common/utils';
 import { refreshPipPackages } from './utils';
 
-export function validatePyprojectToml(toml: tomljs.JsonMap): string | undefined {
-    // 1. Validate package name (PEP 508)
-    if (toml.project && (toml.project as tomljs.JsonMap).name) {
-        const name = (toml.project as tomljs.JsonMap).name as string;
-        // PEP 508 regex: must start and end with a letter or digit, can contain -_., and alphanumeric characters. No spaces allowed.
-        // See https://peps.python.org/pep-0508/
-        const nameRegex = /^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9._-]*[a-zA-Z0-9])$/;
-        if (!nameRegex.test(name)) {
-            return l10n.t('Invalid package name "{0}" in pyproject.toml.', name);
-        }
+export interface PyprojectToml {
+    project?: {
+        name?: string;
+        version?: string;
+    };
+    'build-system'?: {
+        requires?: unknown;
+    };
+}
+export function validatePyprojectToml(toml: PyprojectToml): string | undefined {
+    // 1. Validate required "requires" field in [build-system] section (PEP 518)
+    const buildSystem = toml['build-system'];
+    if (buildSystem && !buildSystem.requires) {
+        // See PEP 518: https://peps.python.org/pep-0518/
+        return l10n.t('Missing required field "requires" in [build-system] section of pyproject.toml.');
+    }
+
+    const project = toml.project;
+    if (!project) {
+        return undefined;
+    }
+
+    const name = project.name;
+    // 2. Validate required "name" field in [project] section (PEP 621)
+    // See PEP 621: https://peps.python.org/pep-0621/
+    if (!name) {
+        return l10n.t('Missing required field "name" in [project] section of pyproject.toml.');
```

Comment:
Love how well formatted this is! Easy to read and understand- thank you! With any of these msgs it aborts pkg install but the env still get created right? might be worth updating the msg to explain that to the user- unless I am missing a msg where that is included